python/lsst/jointcal/utils.py

```diff
@@ -52,41 +58,72 @@ def compute_rms(self, data_refs, visit_list, reference):
+            refcalib = calibs[0]
+            dist_rel, flux_rel, ref_flux_rel, source_rel = self._make_match_dict(refcat, catalogs[1:], calibs,
+                                                                                 refcalib=refcalib)
+            dist_abs, flux_abs, ref_flux_abs, source_abs = self._make_match_dict(reference, catalogs, calibs)
+            dist = MatchDict(dist_rel, dist_abs)
+            flux = MatchDict(flux_rel, flux_abs)
+            ref_flux = MatchDict(ref_flux_rel, ref_flux_abs)
+            source = MatchDict(source_rel, source_abs)
+            return dist, flux, ref_flux, source
+
+        old_cats = [ref.get('src') for ref in data_refs]
+        old_calibs = [ref.get('calexp').getCalib() for ref in data_refs]
+        self.old_dist, self.old_flux, self.old_ref_flux, self.old_source = compute(old_cats, old_calibs)
+
+        # Update coordinates with the new wcs, and get the new Calibs.
         new_cats = []
-        for dataRef in data_refs:
-            new_cats.append(dataRef.get('src'))
-            lsst.afw.table.utils.updateSourceCoords(dataRef.get('wcs').getWcs(), new_cats[-1])
-        visistCatalogs, self.new_relative, self.new_absolute = compute(new_cats)
-
-        self.old_rel_total = rms_total(self.old_relative)
-        self.new_rel_total = rms_total(self.new_relative)
-        self.old_abs_total = rms_total(self.old_absolute)
-        self.new_abs_total = rms_total(self.new_absolute)
-
-        return self.new_rel_total, self.new_abs_total
-
-    def make_plots(self, data_refs, visit_catalogs, old_wcs_list, name='',
-                   interactive=False, per_ccd_plot=False, outdir='.plots'):
+        new_calibs = []
+        for ref in data_refs:
+            new_cats.append(ref.get('src'))
+            wcs = ref.get('wcs')
+            new_calibs.append(wcs.getCalib())
+            # update in-place the object coordinates based on the new wcs
+            lsst.afw.table.utils.updateSourceCoords(wcs.getWcs(), new_cats[-1])
+
+        self.new_dist, self.new_flux, self.new_ref_flux, self.new_source = compute(new_cats, new_calibs)
+
+        self._photometric_rms()
+
+        if self.verbose:
+            print('"photometric factor" for each data ref:')
+            for ref, old, new in zip(data_refs, old_calibs, new_calibs):
+                print(tuple(ref.dataId.values()), new.getFluxMag0()[0]/old.getFluxMag0()[0])
+
+        def rms_total(data):
+            """Compute the total rms across all sources."""
+            total = sum(sum(dd**2) for dd in data.values())
+            n = sum(len(dd) for dd in data.values())
+            return np.sqrt(total/n)
+
+        self.old_dist_total = MatchDict(*(tuple(map(rms_total, self.old_dist))*u.radian).to(u.arcsecond))
+        self.new_dist_total = MatchDict(*(tuple(map(rms_total, self.new_dist))*u.radian).to(u.arcsecond))
+        # TODO: This is wrong!
+        # TODO: The correct calculation is rms(old_ref_flux-old_flux)
```

Comment:
Oh, whoops! That comment was no longer relevant, since I now do all the necessary math in `_photometric_rms()`. Deleted. Those will get returned when I make the unittests work (which will require changing the API: I'll use a namedtuple as the return value of `compute_rms`.